synapse/events/utils.py

```diff
@@ -112,10 +118,16 @@ def add_fields(*fields):
             "kick",
             "redact",
         )
+
+        if room_version.msc2176_redaction_rules:
+            add_fields("invite")
+
     elif event_type == EventTypes.Aliases and room_version.special_case_aliases_auth:
         add_fields("aliases")
     elif event_type == EventTypes.RoomHistoryVisibility:
         add_fields("history_visibility")
+    elif event_type == EventTypes.Redaction and room_version.msc2176_redaction_rules:
+        add_fields("redacts")
```

Comment:
Note that this is really matrix-org/matrix-doc#2174. I suspect further changes are needed to support this although I'm not 100% sure what.

Reply:
Looks like this covers the requirements for MSC2176. Any further changes can be in a separate MSC2174 impl.

Reply:
The only odd thing about this is that currently a redaction event doesn't *have* the redacts keyword, so I think this will generally raise an exception. (Note that the tests somewhat skip this by manually creating an event that does have the proper keyword.)

Reply:
I'm not sure MSC2174 actually requires this.

Reply:
Ah, I think I misread the `add_fields` code, I was afraid it would break unless MSC2174 was implemented, but it should be OK.

MSC2174 doesn't require this, but MSC2176 requires it "assuming that MSC2174 is merged".